zeroconf.py

```diff
@@ -2441,38 +2456,52 @@ def handle_query(self, msg: DNSIncoming, addr: Optional[str], port: int) -> None
+            elif not can_send_to(s, addr):
+                return
+            else:
+                real_addr = addr
+            bytes_sent = s.sendto(packet, 0, (real_addr, port))
+        except Exception as exc:  # TODO stop catching all Exceptions
+            if (
+                isinstance(exc, OSError)
+                and exc.errno == errno.ENETUNREACH
+                and s.family == socket.AF_INET6
+            ):
+                # with IPv6 we don't have a reliable way to determine if an interface actually has IPv6
+                # support, so we have to try and ignore errors.
+                return
+            # on send errors, log the exception and keep going
+            self.log_exception_warning()
+        else:
+            if bytes_sent != len(packet):
+                self.log_warning_once('!!! sent %d out of %d bytes to %r' % (bytes_sent, len(packet), s))
+
+    
     def send(self, out: DNSOutgoing, addr: Optional[str] = None, port: int = _MDNS_PORT) -> None:
         """Sends an outgoing packet."""
         packet = out.packet()
         if len(packet) > _MAX_MSG_ABSOLUTE:
             self.log_warning_once("Dropping %r over-sized packet (%d bytes) %r", out, len(packet), packet)
             return
         log.debug('Sending %r (%d bytes) as %r...', out, len(packet), packet)
-        for s in self._respond_sockets:
+        s = self._listen_socket
+        
+        for so in self._respond_socket_options:
+            s.setsockopt(so["IPProto"],so["SocketOpt"],so["SocketOptValue"])
             if self._GLOBAL_DONE:
                 return
-            try:
-                if addr is None:
-                    real_addr = _MDNS_ADDR6 if s.family == socket.AF_INET6 else _MDNS_ADDR
-                elif not can_send_to(s, addr):
-                    continue
-                else:
-                    real_addr = addr
-                bytes_sent = s.sendto(packet, 0, (real_addr, port))
-            except Exception as exc:  # TODO stop catching all Exceptions
-                if (
-                    isinstance(exc, OSError)
-                    and exc.errno == errno.ENETUNREACH
-                    and s.family == socket.AF_INET6
-                ):
-                    # with IPv6 we don't have a reliable way to determine if an interface actually has IPv6
-                    # support, so we have to try and ignore errors.
-                    continue
-                # on send errors, log the exception and keep going
-                self.log_exception_warning()
-            else:
-                if bytes_sent != len(packet):
-                    self.log_warning_once('!!! sent %d out of %d bytes to %r' % (bytes_sent, len(packet), s))
+            self.send_via_configured_socket(s,packet,addr,port)
+
+        for so in self._respond_sockets:
+            if self._GLOBAL_DONE:
+                return
+            self.send_via_configured_socket(s,packet,addr,port)
```

Comment:
Are we sending through the listening socket (`s`), or are we actually sending through a respond socket (`so`)?

If we are not using `so` in the `for` loop above, we might as well name it `_`, so that it's clear that this is an anonymous variable. But that would also make the whole loop kind of useless I think (we'd just keep sending over the same socket over and over again).

Reply:
That was a typo in code that is not exercise in our use case. Fix on the way.